File: server/src/Serene.API/Common/Results/ApplicationErrors.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

[JsonConverter(typeof(StringEnumConverter))]

public enum AppErrors
{
    UserNotFound,
    UserUpdateError,
    UserAlreadyExists,
    MoodEntryNotReady,
    MoodEntryCompletedAlready,
    AuthInvalidPassword,
    RefreshTokenError,
    ClaimsIdentityNotFound,
    ClaimsEmailNotFound,
    AuthGoogleLoginError,
    AuthGoogleCallbackError,
    InvalidReturnUrl,
}


public static class AppErrorExtensions
{
    public static AppErrors ToAppError(this string error) =>
        Enum.Parse<AppErrors>(error);
    public static string ToJSON(this AppErrors error) =>
        JsonConvert.ToString(error);
}

File: client/src/lib/types/application-errors.ts
export enum AppErrors
{
    UserNotFound = "UserNotFound",
    UserUpdateError = "UserUpdateError",
    UserAlreadyExists = "UserAlreadyExists",
    MoodEntryNotReady = "MoodEntryNotReady",
    MoodEntryCompletedAlready = "MoodEntryCompletedAlready",
    AuthInvalidPassword = "AuthInvalidPassword",
    RefreshTokenError = "RefreshTokenError",
    ClaimsIdentityNotFound = "ClaimsIdentityNotFound",
    ClaimsEmailNotFound = "ClaimsEmailNotFound",
    AuthGoogleLoginError = "AuthGoogleLoginError",
    AuthGoogleCallbackError = "AuthGoogleCallbackError",
    InvalidReturnUrl = "InvalidReturnUrl",
}
